Observer/Frontend/CustomerLoggedIn.php

```diff
@@ -4,28 +4,38 @@
 
 use Sailthru\MageSail\Helper\ClientManager;
 use Sailthru\MageSail\Cookie\Hid as SailthruCookie;
+use Sailthru\MageSail\Helper\Settings as SailthruSettings;
 use Magento\Framework\Event\Observer;
 use Magento\Framework\Event\ObserverInterface;
+use Magento\Newsletter\Model\Subscriber;
 
 class CustomerLoggedIn implements ObserverInterface
 {
 
     private $sailthruCookie;
     private $sailthruClient;
+    private $sailthruSettings;
+    private $subscriber;
 
     public function __construct(
         ClientManager $clientManager,
-        SailthruCookie $sailthruCookie
+        SailthruCookie $sailthruCookie,
+        SailthruSettings $sailthruSettings,
+        Subscriber $subscriber
     ) {
         $this->sailthruCookie = $sailthruCookie;
         $this->sailthruClient = $clientManager;
+        $this->sailthruSettings = $sailthruSettings;
+        $this->subscriber = $subscriber;
     }
 
     public function execute(Observer $observer)
     {
         $customer = $observer->getData('customer');
-        $this->sailthruClient = $this->sailthruClient->getClient(true, $customer->getStore()->getId());
+        $storeId = $customer->getStore()->getId();
+        $this->sailthruClient = $this->sailthruClient->getClient(true, $storeId);
         $sid = $customer->getData('sailthru_id');
+        $newsletterList = $this->sailthruSettings->getNewsletterList($storeId);
```

Comment:
Can you re-arrange these so that customer-related and setting-related vars go in sequential order?